primefaces/src/main/java/org/primefaces/util/CurrencyValidator.java

```diff
@@ -161,9 +161,9 @@ protected Object parseFormat(String value, DecimalFormat formatter) {
      * @return The parsed <code>Number</code> converted to a <code>BigDecimal</code>.
      */
     protected BigDecimal processParsedValue(Number value, DecimalFormat formatter) {
-        BigDecimal decimal = null;
+        BigDecimal decimal;
         if (value instanceof Long) {
-            decimal = BigDecimal.valueOf(((Long) value).longValue());
```

Comment:
Don't we want to keep the long value here?

Reply:
In this case the autoboxing feature (since Java 5) of the JVM do the unboxing for us and improves code readability.